eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/MetricsHandler.java

```diff
@@ -134,15 +132,8 @@ void get(HttpExchange httpExchange) throws IOException {
             String result = JsonUtils.toJSONString(error);
             byte[] bytes = result.getBytes(Constants.DEFAULT_CHARSET);
             httpExchange.sendResponseHeaders(500, bytes.length);
-            out.write(bytes);
-        } finally {
-            if (out != null) {
-                try {
-                    out.close();
-                } catch (IOException e) {
-                    log.warn("out close failed...", e);
-                }
-            }
+
+            log.error(result, e);
```

Comment:
Remove ` log.error(result, e);`? and not remove `out.write(bytes);`

Reply:
because "out" stream went out of scope in catch block. Hence, the code would not compile.

Reply:
> Here you may still have to use try-with-finally, using try-with-resources will be due to the scope of the variable, here in the catch there is out.write(result.getBytes()); this line of code is reported as an error. Do you have a better suggestion?

From Java 9 onwards, I think this could be solved with declaring out stream outside try block :

OutputStream out = httpExchange.getResponseBody();
try (out) {
} catch (Exception e) {
  // use out here.
}

But with Java 8, proposing use of nested try blocks so that out of scope variable could be used.
Also solving compilation issue due to issue#3007 and issue#3008.

Please let me know your inputs.